x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -320,5 +323,17 @@ export function MachineLearningDataVisualizerTableProvider(
         await this.assertTextFieldContents(fieldName, docCountFormatted, exampleCount);
       }
     }
+
+    public async ensureNumRowsPerPage(n: 10 | 25 | 100) {
+      await retry.tryForTime(10000, async () => {
+        await testSubjects.existOrFail('tablePaginationPopoverButton');
```

Comment:
This is working fine at the moment, but will only work as long as there's only one table on the page, otherwise it might pick the wrong pagination control. I'd suggest to make this specific to the data visualizer table by nesting the test subject: `mlDataVisualizerTable > tablePaginationPopoverButton`.
Also, this is used three times in this method, so it could be a local const.